addons/website/models/website.py

```diff
@@ -1062,7 +1062,7 @@ def viewref(self, view_id, raise_if_not_found=True):
             else:
                 domain = [('key', '=', view_id)]
                 order = View._order
-            views = View.with_context(active_test=False).search(domain, order=order)
+            views = View.with_context(active_test=False).search(domain, order=order, limit=1)
```

Comment:
This is not correct, this will stop retrieving all COW views (which are then filtered).
This fact we retrieve all duplicated views is done on purpose and the core behavior of website, I don't know what impact this would have but clearly a no go, it's really a violent change in a low level method of website.

Also, you would need to  rename `views` if it holds only one view